docs/content/guides/developer/advanced/congestion-control.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Congestion Control
+description: Congestion control is the system that limits the rate of transactions writing to a single shared object, preventing the network from becoming overloaded with checkpoints that take too long to execute.
+---
+
+The Sui network’s object-based architecture allows us to process many different user transactions massively in parallel, in a way that’s not possible on most other networks. However, if multiple transactions are all writing to the same shared object, they must execute in sequential order. There is a limit to how many transactions we can process that touch one specific object.
+
+**Congestion control** is the system that limits the rate of transactions writing to a single shared object, preventing the network from becoming overloaded with checkpoints that take too long to execute.
+
+If you see transactions fail with the error `ExecutionCancelledDueToSharedObjectCongestion`, you are observing congestion control at work! Read this document to learn:
+- How transaction space is allocated on a congested shared object.
+- How to bid for priority access to limited space.
+- How to structure your applications and transactions for maximum throughput.
+
+## How congestion control works
+
+Sui's congestion control algorithm runs every time a new batch of sequenced transactions is received from consensus.
+
+1. All transactions are **sorted in order of gas price**.
+
+1. Each transaction's **execution cost is estimated** using the [`TotalGasBudgetWithCap`](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority/shared_object_congestion_tracker.rs#L143) heuristic.
+
+1. The [`SharedObjectCongestionTracker`](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority/shared_object_congestion_tracker.rs) keeps a running tally of how much **per-object congestion budget** is used. If *all* shared objects used by the transaction have enough budget left, the transaction is scheduled. Otherwise it is **deferred** until the next commit.
+
+1. When a transaction is scheduled, it consumes budget for the *mutable* shared objects that it uses. (*Immutable* shared objects don't consume any budget, because multiple immutable uses of a shared object can be executed in parallel.)
+
+### Priority ordering of transactions
+
+Transaction priority is determined solely by **gas price**. If you want your transaction to be assigned access to shared objects ahead of others, you must pay a higher gas price than the others. This is the only way to get priority access.
```

Comment:
Maybe a good place to include a note about SIP45 and the 5xRGP amplification threshold. It's a bit of a type error because congestion control is not technically related to SIP45 but we don't talk about this elsewhere, and this docs seems like the place where folks are most likely to look.

Reply:
Can you propose any wording? I don't know exactly how that works

Reply:
Here's my attempt (and tagging in @mwtian):

- When deciding which transactions to include in a consensus commit, Sui sorts by **gas price** (highest first)
- In addition, if a transaction's gas price is >=5X the current **reference gas price**, the transaction will be submitted to consensus faster than a tx submitted at the same time with a price <5X times the current reference gas price (see [SIP45](https://github.com/sui-foundation/sips/blob/main/sips/sip-45.md) for details)

Thus, if you want a transaction to be processed as quickly as possible, set a price that is at least 5X the RGP.

Reply:
Since this discussion has been resolved, I assume we no longer want to add the details. But if we want to, we can just add a `NOTE: using a higher gas price (> 5 RGP) will also increase the likelihood the transaction gets included in the earliest consensus commit possible.`.